Html pages with css

Mefjuu94 opened the pull request:
03.09.2024
* added validation to static resources
* added docker tests

24.09.2024
* added geocoding -> searching by city name (not finish yet) but it works!
* added missing line to register new user

27.09.2024
* changed some styles with searching results
* changed view of duration of the trip. Now if trip is  > 60 minutes then shows also hours/minutes/seconds instead only minutes and seconds
* added button "add announcement" who collect data (at this point only shows waypoints, and duration via console log).

29.09.2024
* added scripts to manage checkboxes and buttons in map.html
- added text description to specific information about trip
- added right bar where we create a description
of the journey
- added few checkboxes
* added "Trip" class which is responsible for a specific route (announcement)

30.09.2024
* added errorController who redirect to page with exact error.
* added variables to map.html form and map.js to sending all   collect information about trip from javascript.
* moved HttpMethod.POST,"/sendData", /map to authorized requestMatchers.
* added new fields in CustomUser class:
- customUserName
- age
- numbersOfTrips
- numbersOfAnnouncements
also added relations OneToMany -> Trips
* Changed in CustomUserController "print" method name to "collectData". Added Required parameters.
After that Create new Trip and saves in database from previous mentioned javascript.
- added also method "getLoggedInUserName" to get email then get CustomUser who make this annoucement to pass it to new Trip instance because Trip collect id from CustomUser.

03.10.2024
* added sketch of method to update users name
* added line "customUser.setCustomUserName("yourName");" in CustomUserDao to fill field "name". Hibernate won't let me pass without this.
* added Http Post request, "user" and "map" pages to login users only. Have to by fully Authenticated.

09.10.2024
* Added class mapController to create and add to database trip object.
* added builder to Trip class for better class management and cleaner code appearance
* added bunch of icons (png) to main page
* added small describes in error pages, to understanding whats happen.
* added bunch of images (png) for tiles in main menu

///What to do:
1. issue wih logOUT-> logout not working
2. inserting trips from the database as advertisements (page)
3. setting the database relationship user - trip
4. adding yourself as a participant of the trip (functionality + button)
5. creating a user profile page
6. updating user parameters (buttons + functionality)'

What is left for testing purposes and will be removed when no longer needed?
- "old" directory with old templates of pages.
- "successRegister" template -> ony for if mapping works
- "sendData" template-> ony for if mapping works
- "updateName" method in CustomUserController -> leave sketch, I will do it in the future.
- errorStyles.css to fill styles like other (main Page)
- a few commented lines in SecurityConfig (trying resolve problem of logOut)
- loginRegisterStyles -> will be deleted with old templates
- styles  -> will be deleted with old templates

Mefjuu94 committed:
+

Mefjuu94 committed:
*added Map page with css and js
*added About Page
- added validation to static resources
- added docker tests

Mefjuu94 committed:
* modify CustomUserDAO ->back to old version before crash.
* added geocoding -> searching by city name (not finish yet) but it works!
* added missing line to register new user

Mefjuu94 committed:
* created add_announcement.html
* changed some styles with searching results
* changed view of duration of the trip. Now if trip is  > 60 minutes then shows also hours/minutes/seconds instead only minutes and seconds
* added button "add announcement" who collect data (at this point only shows waypoints, and duration via console log).

Mefjuu94 committed:
* modifying the appearance of the map page (added some styles)
* added scripts to manage checkboxes and buttons in map.html
- added text description to specific information about trip
- added right bar where we create a description
of the journey
- added few checkboxes
* added "Trip" class which is responsible for a specific route (announcement)
added "TripDAO" to management Trip Class in database (not finished yet)

Mefjuu94 committed:
* added variables witch passes from java to javascipt (in Controller)
- after click on "save annoucement" should console log "moje koordynaty" in browser toverify if array has sucessfuly passed.
* added pass variables from JavaScript to Java by /sendData action, printed it, and go to sendData.html to verify succes of operation.
* added (temporary) HttpMethod.POST,"/sendData" to Http request to requestMatchers to be able fast testing (In future only authorized CustomUser)

Mefjuu94 committed:
* added error html pages
* added errorController who redirect to page with exact error.
* added variables to map.html form and map.js to sending all   collect information about trip from javascript.
* moved HttpMethod.POST,"/sendData", /map to authorized requestMatchers.
* added new fields in CustomUser class:
- customUserName
- age
- numbersOfTrips
- numbersOfAnnouncements
also added relations OneToMany -> Trips
* Changed in CustomUserController "print" method name to "collectData". Added Required parameters.
After that Create new Trip and saves in database from previous mentioned javascript.
- added also method "getLoggedInUserName" to get email then get CustomUser who make this annoucement to pass it to new Trip instance because Trip collect id from CustomUser.

* added sketch of method to update users name
* added line "customUser.setCustomUserName("yourName");" in CustomUserDao to fill field "name". Hibernate won't let me pass without this.
* added Http Post request, "user" and "map" pages to login users only. Have to by fully Authenticated.

Mefjuu94 committed:
* changed the overall look to "one page" view, most of the information is on the main page, access to functions for no logged users is grayed out (if click then pop out modal window to login), and for logged images have colors (means available) access is protected by authentication from thymeleaf template in html.
* Added class mapController to create and add to database trip object.
* added builder to Trip class for better class management and cleaner code appearance
* added bunch of icons (png) to main page
* added small describes in error pages, to understanding whats happen.
* added bunch of images (png) for tiles in main menu

///What to do:
1. issue wih logOUT-> logout not working
2. inserting trips from the database as advertisements (page)
3. setting the database relationship user - trip
4. adding yourself as a participant of the trip (functionality + button)
5. creating a user profile page
6. updating user parameters (buttons + functionality)'

What is left for testing purposes and will be removed when no longer needed?
- "old" directory with old templates of pages.
- "successRegister" template -> ony for if mapping works
- "sendData" template-> ony for if mapping works
- "updateName" method in CustomUserController -> leave sketch, I will do it in the future.
- errorStyles.css to fill styles like other (main Page)
- a few commented lines in SecurityConfig (trying resolve problem of logOut)
- loginRegisterStyles -> will be deleted with old templates
- styles  -> will be deleted with old templates

blazejprzyluski commented on src/main/java/TTT/controller/CustomUserController.java:
```diff
@@ -0,0 +1,34 @@
+    }
+
+    @GetMapping("/map")
+    public String getMapPage() {
+        return "map";
+    }
+
+    @GetMapping("/announcement")
+    public String getAnnouncementsPage() {
+        return "announcement";
+    }
```
To be moved from this controller

Mefjuu94 replied:
Done. Created new controller Class to announcements.

blazejprzyluski commented on src/main/java/TTT/controller/MapController.java:
```diff
@@ -0,0 +1,81 @@
+                              @RequestParam("isCheckedGroup") String isCheckedGroup,
+                              @RequestParam("amountOfPeopleInGroup") String amountOfPeopleInGroup) {
+
+        System.out.println("Otrzymano dane: ");
+        System.out.println("zmienna 1: " + waypoints);
+        System.out.println("zmienna 2: " + coordinatesOfTrip);
+        System.out.println("Zmienna 3: " + allRouteDuration);
+
+        System.out.println("Zmienna 4 opis: " + description);
+        System.out.println("Zmienna 5 driver?: " + driverCheck);
+        System.out.println("Zmienna 6 liczba osób w samochodzie: " + amountOfPeopleDriver);
+        System.out.println("Zmienna 7 zwierzaki?: " + isCheckedAnimals);
+        System.out.println("Zmienna 8 grupa zamknięta?: " + isCheckedGroup);
+        System.out.println("Zmienna 9 ile osób w grupie: " + amountOfPeopleInGroup);
```
To be removed

Mefjuu94 replied:
Done.

blazejprzyluski commented on src/main/java/TTT/controller/MapController.java:
```diff
@@ -0,0 +1,81 @@
+        return "sendData";
+    }
+
+    public String getLoggedInUserName() {
```
Should be private?

Mefjuu94 replied:
Yes. Corrected.

blazejprzyluski commented on src/main/java/TTT/controller/SecurityController.java:
```diff
@@ -27,10 +27,17 @@ public String getRegisterPage(Model model){
+        if (bindingResult.hasErrors()) {
             return "security/register";
         }
+        if (customUser.getPassword().length() < 8){
+            return "index";
+        }
```
Let's discuss how to validate password on next meeting

Mefjuu94 replied:
ok...

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -16,22 +16,36 @@ public class CustomUserDAO {
+        if (findCustomUser(customUser.getEmail()) != null) {
+            return false;
+        }
+        if (customUser.getPassword().length() < 7) {
```
In the controller you are checking for length of 8, here 7

Mefjuu94 replied:
My bad. Corrected.

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -16,22 +16,36 @@ public class CustomUserDAO {
+
+        Session session = null;
+        Transaction transaction = null;
         try {
```
session can be created in a try with resources

Mefjuu94 replied:
ok! Done.

blazejprzyluski commented on src/main/java/TTT/databaseUtils/CustomUserDAO.java:
```diff
@@ -41,10 +55,74 @@ public CustomUser findCustomUser(String email) {
+        }
+
+        Session session = null;
+        try {
```
Session can be created in a try with resources

Mefjuu94 replied:
Done.

blazejprzyluski commented on src/main/java/TTT/databaseUtils/TripDAO.java:
```diff
@@ -0,0 +1,88 @@
+
+        Session session = null;
+        Transaction transaction = null;
+        try {
```
Try with resources for session

Mefjuu94 replied:
Done.

blazejprzyluski commented on src/main/java/TTT/trips/Trip.java:
```diff
@@ -0,0 +1,178 @@
+    private boolean closedGroup;
+    private int amountOfClosedGroup;
+
+    private boolean DriverPeople;
```
Drivers sounds better

Mefjuu94 replied:
better be "peopleInTheCar" because this is about how many people I can take with my car

blazejprzyluski commented on src/main/resources/static/images/1 (2).jpg:
Repeats with background file

Mefjuu94 committed:
What was DONE:
* inserting trips from the database as advertisements (page)
* inserting ID of trips and ID of CustomUser to trips_participants table
* setting the database relationship user - trip
* adding yourself as a participant of the trip (functionality + button)
* creating a user profile page
* added few methods to update updating user parameters (buttons + functionality) in user profile page
* Added AnnouncementController to control any announcement, and creates page. (will be updated...)
* fixed things that had a comment in the last pull request

* Deleted unnecessary staff like images, directories, and old files.

What still to do:
1. issue wih logOUT-> logout not working
2. verification already in view (via .js) whether we have signed up for a trip or which trips are ours (we are the owners) because now had only messaged in  java that we are owner or already we are participant of trip.

Mefjuu94 committed:
* Creating user page with functionality to change some fields like name, age, ect. (page + functionality)
* Modifying CustomUserDAO methods about change fields "email" "username" "city". Now It is a one method instead of 3 different because of "switch" statement
* Added Styles for errors pages, and errors pages now redirect to mainPage after count 5 seconds.
* Create "Success Page" -> after some changes like rename, or create Announcement ect. page redirect to main page after count 5 secounds like in errors.
* added icons "owner" "add" "ok" at announcement page to know if user ar already on this trip, or if he can join to trip, or it's owner of trip

Mefjuu94 committed:
* Creating user page with functionality to change some fields like name, age, ect. (page + functionality)
* Modifying CustomUserDAO methods about change fields "email" "username" "city". Now It is a one method instead of 3 different because of "switch" statement
* Added Styles for errors pages, and errors pages now redirect to mainPage after count 5 seconds.
* Create "Success Page" -> after some changes like rename, or create Announcement ect. page redirect to main page after count 5 secounds like in errors.
* added icons "owner" "add" "ok" at announcement page to know if user ar already on this trip, or if he can join to trip, or it's owner of trip

Mefjuu94 committed:
Merge branch 'main' into HtmlPagesWithCss
